internal/pkg/agent/cmd/run.go

```diff
@@ -715,12 +716,23 @@ func initTracer(agentName, version string, mcfg *monitoringCfg.MonitoringConfig)
 
 func setupMetrics(
 	logger *logger.Logger,
-	operatingSystem string,
 	cfg *monitoringCfg.MonitoringConfig,
 	tracer *apm.Tracer,
 	coord *coordinator.Coordinator,
 ) (*reload.ServerReloader, error) {
-	if err := report.SetupMetrics(logger, agentName, version.GetDefaultVersion()); err != nil {
+	ephemeralID, err := uuid.NewV4()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate ephemeral id: %w", err)
+	}
+
+	if err = report.SetupMetricsOptions(report.MetricOptions{
+		Name:           agentName,
+		Version:        version.GetDefaultVersion(),
+		EphemeralID:    ephemeralID.String(),
+		Logger:         logger,
+		SystemMetrics:  monitoringLib.Default.GetOrCreateRegistry("system"),
+		ProcessMetrics: monitoringLib.Default.GetOrCreateRegistry("beat"),
+	}); err != nil {
```

Comment:
This is here to fix the deprecation of report.SetupMetrics ? 

Would it be possible to take this out of this PR and substitute with a `//nolint` comment ? 

Asking mostly because the way of removing the deprecation warning is not documented and I am not comfortable just pasting here the implementation of `report.SetupMetrics` with an ephemeral UUID